apps/web/src/modules/create-proposal/components/TransactionForm/Migration/prepareMigrationDeploy.ts

```diff
@@ -0,0 +1,190 @@
+    abi: governorAbi,
+    address: governor as Address,
+    chainId: chain.id,
+  }
+
+  const treasuryContractParams = {
+    abi: treasuryAbi,
+    address: treasury as Address,
+    chainId: chain.id,
+  }
+
+  const contractData = await readContracts({
+    allowFailure: false,
+    contracts: [
+      { ...tokenContractParams, functionName: 'name' },
+      { ...tokenContractParams, functionName: 'symbol' },
+      { ...tokenContractParams, functionName: 'getFounders' },
+      { ...tokenContractParams, functionName: 'totalSupply' },
+      { ...metadataContractParams, functionName: 'contractImage' },
+      { ...metadataContractParams, functionName: 'description' },
+      { ...metadataContractParams, functionName: 'contractURI' },
+      { ...auctionContractParams, functionName: 'duration' },
+      { ...auctionContractParams, functionName: 'reservePrice' },
+      { ...governorContractParams, functionName: 'votingDelay' },
+      { ...governorContractParams, functionName: 'votingPeriod' },
+      { ...governorContractParams, functionName: 'proposalThresholdBps' },
+      { ...governorContractParams, functionName: 'quorumThresholdBps' },
+      { ...governorContractParams, functionName: 'vetoer' },
+      { ...treasuryContractParams, functionName: 'delay' },
+    ] as const,
+  })
+
+  const [
+    name,
+    symbol,
+    existingFounders,
+    totalSupply,
+    daoImage,
+    description,
+    contractURI,
+    duration,
+    reservePrice,
+    votingDelay,
+    votingPeriod,
+    proposalThresholdBps,
+    quorumThresholdBps,
+    vetoer,
+    timelockDelay,
+  ] = unpackOptionalArray(contractData, 13)
+
+  const founderParams = existingFounders
+    ? [
+        zeroFounder,
+        ...existingFounders.map((x) => ({
+          wallet: x.wallet,
+          ownershipPct: BigInt(x.ownershipPct),
+          vestExpiry: BigInt(x.vestExpiry),
+        })),
+      ]
+    : [zeroFounder]
+
+  const abiCoder = new ethers.utils.AbiCoder()
+
+  const merkleMinterSettingsHex = encodeAbiParameters(
+    [
+      { name: 'mintStart', type: 'uint64' },
+      { name: 'mintEnd', type: 'uint64' },
+      { name: 'pricePerToken', type: 'uint64' },
+      { name: 'merkleRoot', type: 'bytes32' },
+    ],
+    [
+      BigInt(0), // can it just be 0?
+      18446744073709551615n, //  max
+      BigInt(0),
+      merkleRoot,
+    ]
+  )
+
+  const initialMinter = L2_DEPLOYMENT_ADDRESSES[targetChainId].MERKLERESERVEMINTER // figure it out based on chain going to
+  const tokenParamsHex = encodeAbiParameters(
+    [
+      { name: 'name', type: 'string' },
+      { name: 'symbol', type: 'string' },
+      { name: 'reservedUntilTokenId', type: 'uint256' },
+      { name: 'initialMinter', type: 'address' },
+      { name: 'initialMinterData', type: 'bytes' },
+    ],
+    [name!, symbol!, totalSupply!, initialMinter, merkleMinterSettingsHex]
+  )
+
+  const propertyMetadataParamsHex = abiCoder.encode(
+    ['string', 'string', 'string', 'string'],
```

Comment:
these params should actually be encoded as structs. pull this from the ABI instead of hardcoding here to get the right values

Reply:
these don't exist in the Manager ABI - should i be pulling it from the IMetadata ABI for example? That's the only place I see the actual structs